woocommerce/woocommerce
# Cache Marketplace search results

### Changes proposed in this Pull Request:

Closes [WCCOM#18292](https://github.com/Automattic/woocommerce.com/issues/18292). The 100 most-recent search/browse/features/category-list results in the Marketplace are cached into a `Map` so that if you do the same browse/search again you get instant results (and don't hit our API) until you navigate away or actually refresh.



### How to test the changes in this Pull Request:

1. Check out this branch.
2. Recompile assets e.g. `cd plugins/woocommerce-admin && pnpm start`
3. In your test environment, visit WooCommerce > Extensions
4. Browse and search. Notice (e.g. using your browser's debug > network panel) that repeat searches/browses don't result in repeat API hits.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

Memory-cache In-App Marketplace search, browse, and category API hits, for performance.

</details>
